File: src/models/VendorQuote.ts
import type { VendorQuoteStatus } from "@/types/auth";

export type VendorQuoteRow = {
  id: string;
  vendor_id: string;
  request_id: string;
  amount: number;
  timeline: string;
  notes: string;
  attachment_url: string | null;
  status: VendorQuoteStatus;
  admin_notes: string | null;
  risk_notes: string | null;
  is_recommended: boolean;
  execution_fit_score: number | null;
  reviewed_at: string | null;
  reviewed_by: string | null;
  created_at: string;
  updated_at: string;
};

export type VendorQuote = {
  id: string;
  vendorId: string;
  requestId: string;
  amount: number;
  timeline: string;
  notes: string;
  attachmentUrl: string | null;
  status: VendorQuoteStatus;
  adminNotes: string | null;
  riskNotes: string | null;
  isRecommended: boolean;
  executionFitScore: number | null;
  reviewedAt: string | null;
  reviewedBy: string | null;
  createdAt: string;
  updatedAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/VendorQuote.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class VendorQuote(
    val id: String,
    @SerializedName("vendor_id") val vendorId: String,
    @SerializedName("request_id") val requestId: String,
    val amount: Double,
    val timeline: String,
    val notes: String,
    @SerializedName("attachment_url") val attachmentUrl: String? = null,
    val status: String = "pending",
    @SerializedName("admin_notes") val adminNotes: String? = null,
    @SerializedName("is_recommended") val isRecommended: Boolean = false,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("updated_at") val updatedAt: String? = null
)
